src/authentication/MobileRouter.jsx

```diff
@@ -1,39 +1,131 @@
-import React from 'react'
+import React, { Component } from 'react'
 import { Router, withRouter } from 'react-router'
-
+import Proptypes from 'prop-types'
 import Authentication from './src/Authentication'
 import Revoked from './src/Revoked'
 import { logException } from 'drive/lib/reporter'
+import {
+  readState,
+  secretExchange,
+  getAccessToken,
+  readSecret,
+  clearSecret,
+  clearState,
+  checkExchangedInformations,
+  checkIfOnboardingLogin
+} from './src/utils/onboarding'
+import {
+  onboardingInformationsPropTypes,
+  onboardingPropTypes
+} from './OnboardingPropTypes'
+class MobileRouter extends Component {
+  async doOnboardingLogin(receivedState, code, cozy_url, history) {
+    const localState = await readState()
+    const localSecret = await readSecret()
+
+    try {
+      if (localState !== receivedState) {
+        throw new Error('States are not equals')
+      }
+
+      const clientInfo = await secretExchange(
+        localSecret,
+        cozy_url,
+        this.props.client
+      )
+
+      const { onboarding_secret, onboarding_state } = clientInfo
+      if (
+        !checkExchangedInformations(
+          localSecret,
+          onboarding_secret,
+          localState,
+          onboarding_state
+        )
+      )
+        throw new Error('exchanged informations are not good')
 
-const MobileRouter = ({
-  history,
-  appRoutes,
-  isAuthenticated,
-  isRevoked,
-  onAuthenticated,
-  onLogout,
-  appIcon
-}) => {
-  if (!isAuthenticated) {
-    return (
-      <Authentication
-        router={history}
-        onComplete={onAuthenticated}
-        onException={logException}
-        appIcon={appIcon}
-      />
-    )
-  } else if (isRevoked) {
-    return (
-      <Revoked
-        router={history}
-        onLogBackIn={onAuthenticated}
-        onLogout={onLogout}
-      />
-    )
-  } else {
-    return <Router history={history}>{appRoutes}</Router>
+      const getTokenRequest = await getAccessToken(
+        clientInfo,
+        cozy_url,
+        code,
+        this.props.client
+      )
+      const token = await getTokenRequest.json()
+      if (getTokenRequest.status !== 200) {
+        throw new Error('token.error')
+      }
+
+      await this.props.onAuthenticated({
+        url: `https://${cozy_url}`,
```

Comment:
Nit, but also kind of not nit: I think we should rename this variable. Not only is it snake cased and the linter should be yelling at it, but it's also not a fully formed URL and we have to add the protocol here and there. I think `instanceName` or `instanceDomain` would be better.